Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +23,61 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    return (current_ < 0);
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+        double totalHoursRemaining;
+
+        if(isCharging())
+        {
+            totalHoursRemaining = totalAmpHoursUsed() / current_ * -1;
+        }
+        else
+        {
+            totalHoursRemaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / current_;
+        }
+
+        while(totalHoursRemaining > 24)
+        {
+            totalHoursRemaining -= 24;
```

Comment:
What is the point of this check? Seems like magic, might benefit from a comment.